src/node/waku_node.py

```diff
@@ -186,3 +255,47 @@ def is_nwaku(self):
 
     def is_gowaku(self):
         return "go-waku" in self.image
+
+    def parse_rln_credentials(self, default_args, is_registration):
+        rln_args = {}
+        keystore_path = None
+
+        creds_f = open(default_args["rln-creds-source"])
+
+        imported_creds = json.load(creds_f)
+        selected_id = default_args["rln-creds-id"]
+
+        if len(imported_creds) < 4 or any(value is None for value in imported_creds.values()):
+            return rln_args, False, keystore_path
+
+        eth_private_key = select_private_key(imported_creds, selected_id)
+
+        if self.is_nwaku():
```

Comment:
can we have an else in case the image is not nwaku so we handle that case as well?
maybe with a log line or error

Reply:
I've added raise exception
https://github.com/waku-org/waku-interop-tests/pull/16/commits/932707cb016ee8c82509ea2783e445f2abbcc2e0